app/models/validations/financial_validations.rb

```diff
@@ -58,4 +58,27 @@ def validate_tshortfall(record)
       record.errors.add :tshortfall, I18n.t("validations.financial.hbrentshortfall.outstanding_no_benefits")
     end
   end
+
+  def validate_rent_amount(record)
+    if record.brent.present? && record.tshortfall.present? && record.brent < record.tshortfall * 2
+      record.errors.add :brent, I18n.t("validations.financial.rent.less_than_double_shortfall", tshortfall: record.tshortfall * 2)
+      record.errors.add :tshortfall, I18n.t("validations.financial.tshortfall.more_than_rent")
+    end
+
+    if record.scharge.present? && record.weekly_value(record.scharge).present?
+      if record.this_landlord?
+        if !record.weekly_value(record.scharge).between?(0, 55) && record.is_general_needs?
+          record.errors.add :scharge, I18n.t("validations.financial.rent.scharge.this_landlord.general_needs")
+        elsif !record.weekly_value(record.scharge).between?(0, 280) && record.is_supported_housing?
+          record.errors.add :scharge, I18n.t("validations.financial.rent.scharge.this_landlord.supported_housing")
+        end
+      elsif record.other_landlord?
+        if !record.weekly_value(record.scharge).between?(0, 45) && record.is_general_needs?
+          record.errors.add :scharge, I18n.t("validations.financial.rent.scharge.other_landlord.general_needs")
+        elsif !record.weekly_value(record.scharge).between?(0, 165) && record.is_supported_housing?
+          record.errors.add :scharge, I18n.t("validations.financial.rent.scharge.other_landlord.supported_housing")
+        end
+      end
+    end
```

Comment:
There's a smaller function inside this function, struggling to get out. I'll find it in a bit